### PR TITLE
Adding app-sre pipeline scripts

### PR DESCRIPTION
**Pull request by ravitri**
Adding app-sre pipeline scripts as per https://issues.redhat.com/browse/OSD-3807. Reference is with aws-account-operator..

### CODE REVIEW

File: hack/generate-operator-bundle.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import datetime
+import os
+import sys
+import yaml
+import shutil
+
+# This script will append the current number of commits given as an arg
+# (presumably since some past base tag), and the git hash arg for a final
+# version like: 0.1.189-3f73a592
+VERSION_BASE = "0.2"
```
Comment (mrbarge) [inline hack/generate-operator-bundle.py:21]: Maybe we can start at 0.1

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+objects:
+- apiVersion: operators.coreos.com/v1alpha1
+  kind: CatalogSource
+  metadata:
+    name: managed-upgrade-operator-catalog
+  spec:
+    sourceType: grpc
+    image: ${REGISTRY_IMG}:${CHANNEL}-${IMAGE_TAG}
+    displayName: managed-upgrade-operator Registry
+    publisher: SRE
```
Comment (mrbarge) [inline hack/olm-registry/olm-artifacts-template.yaml:23]: We should maybe be setting an affinity for infra nodes here, as suggested by OSD-4113. See cloud-ingress-operator as an example.

**Comment by sedroche** [comment]
Looks good to me but I'm not hugely familiar with the process. I would appreciate a quick run through if you have time

File: hack/generate-operator-bundle.py (hunk 1 above)
Comment (ravitri) [reply hack/generate-operator-bundle.py:21]: Ahhh!! I swear I had changed this. Wonder why it was not changed.. Thanks to catch it though.. Will fix this.

File: hack/olm-registry/olm-artifacts-template.yaml (hunk 2 above)
Comment (ravitri) [reply hack/olm-registry/olm-artifacts-template.yaml:23]: Nice point Matt.. As per https://github.com/openshift/cloud-ingress-operator/commit/227c095f11de23183bcdc60d21345a296f0e19e0#diff-e12c68acf5db1ed8c169f6a9c8a722a6 I see that requiredDuringSchedulingIgnoredDuringExecution is used to schedule operator pod on infra node. Whereas, https://github.com/openshift/rbac-permissions-operator/pull/67/files#diff-5e5f804a8ee2fb8c81711b728eb5a21d is still a discussion where its suggested to use preferredDuringSchedulingIgnoredDuringExecution. I think its better to have a separate JIRA card for this and address it after this is decided. Sounds ok?

**Comment by ravitri** [comment]
@sedroche - Sure.. to be honest, I myself am struggling to understand it all because of so many pieces involved but I can have a session for it for team to have an overview.

File: hack/olm-registry/olm-artifacts-template.yaml (hunk 2 above)
Comment (mrbarge) [reply hack/olm-registry/olm-artifacts-template.yaml:23]: Sure, a separate card sounds fine to me, let's get clarification on required vs preferred and then implement.

**Pull request merged**